### DISCUSSION TITLE
Losing pivot data

### DISCUSSION THREAD

**Discussion by vanrijs** [livewire/livewire#1485] [4. General]
Hi!

I define a variable in mount() that has pivot data. Im able to use this data in my view but after a re-render, my pivot data in the variable is gone.

Im able to fix this by defines this variable in render() instead of mount() but I don't think this should be necessary.

So first load my data set looks like this:
`  #original: array:6 [▼
    "id" => 1
    "name" => "Cancel at"
    "field" => "cancel_at"
    "pivot_template_id" => 1
    "pivot_field_id" => 1
    "pivot_validate" => "Y"
  ]`

After a click or input (re-render), my data set looks like this:
`  #original: array:3 [▼
    "id" => 1
    "name" => "Cancel at"
    "field" => "cancel_at"
  ]`

Any ideas? Thanks in advance! :) 

Raymond

**Comment by roni-estein** [comment]
I understand skipping mount is only for simple assigns. If you need to load relations you’ll probably want to do that in the mount or hydrated functions
